File: onedayAlgorithm/2022/02_Feburary/5th/탐색/4-1.py
import sys
sys.stdin = open("input.txt", "r")

N = int(input())
commands = list(input().split())

console = {"R": (0, 1), "L": (0, -1), "U": (-1, 0), "D": (1, 0)}
now = [0, 0]

for command in commands:
    r, c = console[command]
    if N < (now[0] + r) or (now[0] + r) < 0 or N < (now[1] + c) or (now[1]) + c < 0:
        continue
    now[0] += r
    now[1] += c

print(now[0]+1, now[1]+1)

File: onedayAlgorithm/2022/02_Feburary/5th/탐색/4-1.cpp
#include <bits/stdc++.h>
using namespace std;

int n;
string commands;
int r = 1, c = 1;

int dr[4] = {0, 0, -1, 1};
int dc[4] = {1, -1, 0, 0};
char console[4] = {'R', 'L', 'U', 'D'};

int main(void) {
    freopen("input.txt", "r", stdin);
    cin >> n;
    cin.ignore();
    getline(cin, commands);

    for (int i = 0; i < commands.size(); i++) {
        char command = commands[i];
        int nr = -1, nc = -1;
        for (int j = 0; j < 4; j++) {
            if (command == console[j]) {
                nr = r + dr[j];
                nc = c + dc[j];
            }
        }
        if (nr < 1 || nc < 1 || nr > n || nc > n) continue;
        r = nr;
        c = nc;
    }
    cout << r << ' ' << c << '\n';
    return 0;
}
